### PR TITLE
Fix: Cancel tag edit when pressing esc & Tags duplicate names

### LINKED ISSUE #11290: Bug: `Esc` cancels Tag edition rather than closing the Settings
### Description

`Esc` cancels Tag edition rather than closing the Settings 

### Steps To Reproduce

1. Go to `Advanced Settings` and modify a tag.
2. Press `Esc` key, the settings close.

### Steps to fix

- Pressing `Esc` should escape only the edition of the tag, not close the 

### Files Version

Version 2.4.33.0

### Windows Version

W10 Home 22H2 19045.2251

### Log file

None

### LINKED ISSUE #11291: Bug: Tags can have similar names
### Description

Tags can currently share the same name, which is prone to confusion.

### Steps To Reproduce

1. Go to Avanced Settings and modify the name of two tags so they have the same.
2. The tags can be exactly similar, which is unclear.

### Steps to fix

- Tags should not be able to have the same name.
- Disable create and save button.

### Files Version

Version 2.4.33.0

### Windows Version

W10 Home 22H2 19045.2251

### Log file

None

### PR DESCRIPTION
**Pull request by ferrariofilippo**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #11290 
- Closes #11291 

**Notes**
- I also removed lines @ 77 and @ 85 which were not really needed

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

### CODE REVIEW

File: src/Files.App/Views/SettingsPages/Advanced.xaml [hunk 1]
```diff
@@ -234,6 +234,9 @@
 							<ListView.ItemTemplate>
 								<DataTemplate xmlns:vm="using:Files.Backend.ViewModels.FileTags" x:DataType="vm:ListedTagViewModel">
 									<Grid ColumnSpacing="16">
+										<Grid.KeyboardAccelerators>
+											<KeyboardAccelerator Key="Escape" Invoked="KeyboardAccelerator_Invoked" />
```
Comment (yaira2) [inline src/Files.App/Views/SettingsPages/Advanced.xaml:238]: Should this be moved to the text box?

Comment (ferrariofilippo) [reply src/Files.App/Views/SettingsPages/Advanced.xaml:238]: This way you can also cancel color change pressing `Esc`

Comment (yaira2) [reply src/Files.App/Views/SettingsPages/Advanced.xaml:238]: I don't think that's needed, users mainly expect to be able to cancel from the text box.

Comment (ferrariofilippo) [reply src/Files.App/Views/SettingsPages/Advanced.xaml:238]: Yes, but moving that to `TextBox` would close settings if the user hits `Esc` when editing color

Comment (yaira2) [reply src/Files.App/Views/SettingsPages/Advanced.xaml:238]: Ignore what I said, I just realized this is only in the list view item template. I thought it was on the whole page.

File: src/Files.App/Views/SettingsPages/Advanced.xaml.cs [hunk 2]
```diff
@@ -105,5 +100,22 @@ private bool IsNameValid(string name)
...
+		private void KeyboardAccelerator_Invoked(KeyboardAccelerator sender, KeyboardAcceleratorInvokedEventArgs args)
+		{
+			if (args.KeyboardAccelerator.Key is VirtualKey.Escape)
+			{
+				AbortEdit();
+				args.Handled = true;
+			}
+		}
+
+		private void AbortEdit()
```
Comment (yaira2) [inline src/Files.App/Views/SettingsPages/Advanced.xaml.cs:113]:
```suggestion
		private void EndEdit()
```

File: src/Files.App/Views/SettingsPages/Advanced.xaml (hunk 1 above)
Comment (ferrariofilippo) [reply src/Files.App/Views/SettingsPages/Advanced.xaml:238]: I believe the user would expect to exit `Edit Mode` and then exit settings pressing another time `Esc`

Comment (ferrariofilippo) [reply src/Files.App/Views/SettingsPages/Advanced.xaml:238]: > Ignore what I said, I just realized this is only in the list view item template. I thought it was on the whole page.

Ok

**Comment by ferrariofilippo** [comment]
@yair100 can I push here also a fix to #11291 or should I open a different PR (is just two lines)

**Comment by yair100** [comment]
> @yair100 can I push here also a fix to #11291 or should I open a different PR (is just two lines)

Might as well combine it 👍

**Review by yair100 — APPROVED** [review]
LGTM

**Pull request merged**